### PR TITLE
[DESKTOP/API] Fix linter errors

### LINKED ISSUE #65: Choose which typescript rules we want

### PR DESCRIPTION
**Pull request by jvnm-dev**
This commit fixes all linter errors.

### CODE REVIEW

**Comment by jvnm-dev** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>Desktop changes but no cypress tests!</td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>Please avoid big changes all at once unless it's absolutely necessary!</td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>You should add an entry into CHANGELOG.md!</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>Hi there, thanks for contributing!</td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against b3985b7d2f5edee24773eba42d36412ddd7aa15a
</p>

**Comment by jvnm-dev** [comment]
> ⚠️	Please avoid big changes all at once unless it's absolutely necessary!

You are not my mom

File: projects/api/__tests__/setup.ts [hunk 1]
```diff
@@ -13,5 +13,5 @@ afterEach(async () => {
 })
 
 afterAll(async () => {
```
Comment (Sylv11) [inline projects/api/__tests__/setup.ts]: No `await`? Remove the `async`?

File: projects/api/__tests__/specs/auth/create.ts [hunk 2]
```diff
@@ -30,7 +30,7 @@ describe('POST /auth/create', () => {
         // Simulating that the email is already taken
         // by creating the user before
         const repository: Repository<User> = getRepository(User)
-        const user = await repository.create(johnDoe)
+        const user = repository.create(johnDoe)
```
Comment (Sylv11) [inline projects/api/__tests__/specs/auth/create.ts]: Type of `user`

File: projects/api/__tests__/specs/auth/login.ts [hunk 3]
```diff
@@ -8,17 +8,17 @@ import { johnDoe } from '../../mocks/user'
...
-    password?: string,
+    email: string;
+    password?: string;
 }
 
 describe('POST /auth/login', () => {
     it('Login successfully', async () => {
         const repository: Repository<User> = getRepository(User)
-        const saltRounds: number = parseInt(`${process.env.SALT_ROUNDS}`, 10)
+        const saltRounds: number = parseInt(process.env.SALT_ROUNDS ?? '10', 10)
```
Comment (Sylv11) [inline projects/api/__tests__/specs/auth/login.ts:18]: Type of `password`

File: projects/api/__tests__/specs/auth/login.ts [hunk 4]
```diff
@@ -8,17 +8,17 @@ import { johnDoe } from '../../mocks/user'
...
         const password = await bcrypt.hash(johnDoe.password, saltRounds)
 
-        const user = await repository.create({
+        const user = repository.create({
```
Comment (Sylv11) [inline projects/api/__tests__/specs/auth/login.ts]: Type

File: projects/api/__tests__/specs/auth/login.ts [hunk 5]
```diff
@@ -57,7 +57,7 @@ describe('POST /auth/login', () => {
 
     it('Login unsuccessfully (incorrect password)', async () => {
         const repository: Repository<User> = getRepository(User)
-        const user = await repository.create(johnDoe)
+        const user = repository.create(johnDoe)
```
Comment (Sylv11) [inline projects/api/__tests__/specs/auth/login.ts]: T-Y-P-E

File: projects/api/src/controllers/authController.ts [hunk 6]
```diff
@@ -60,12 +60,12 @@ export const authController = {
...
     create: async (data: IUser): Promise<object | BoomType> => {
         // Hashing password
-        const saltRounds: number = parseInt(`${process.env.SALT_ROUNDS}`, 10)
+        const saltRounds: number = parseInt(process.env.SALT_ROUNDS ?? '10', 10)
         const clearPassword: string = data.password
         data.password = await bcrypt.hash(clearPassword, saltRounds)
 
         try {
-            await getManager().transaction(async transactionalEntityManager => {
+            await getManager().transaction(async (transactionalEntityManager: EntityManager) => {
```
Comment (Sylv11) [inline projects/api/src/controllers/authController.ts]: Do we know what is the returned type of the async function?

File: projects/desktop/src/components/atoms/common/Alert.tsx [hunk 7]
```diff
@@ -6,9 +6,16 @@ import { getRawTheme } from '../../../redux/helpers/theme'
...
+interface IAlertProps {
+    bgColor?: PropsValue;
+    color?: PropsValue;
+    borderColor?: PropsValue;
+}
+
 const AlertBox = styled.div<IAlertColor>`
-    background: ${props => props.bgColor};
-    color: ${props => props.color};
+    background: ${(props: IAlertProps): PropsValue => props.bgColor};
```
Comment (Sylv11) [inline projects/desktop/src/components/atoms/common/Alert.tsx]: Is it not better to put `props.bgColor ?? ''` and remove the `| undefined` of the `PropsValue` type?

File: projects/desktop/src/containers/LoginContainer.tsx [hunk 8]
```diff
@@ -42,7 +42,7 @@ type IProps = IDispatchProps
  */
 
 const mapDispatchToProps = (dispatch: Function): IDispatchProps => ({
-    setTheme: (theme: string): Promise<Function> => dispatch(setTheme(theme)),
+    setTheme: async (theme: string): Promise<Function> => dispatch(setTheme(theme)),
```
Comment (Sylv11) [inline projects/desktop/src/containers/LoginContainer.tsx:45]: We need `async` now? Was it not handled by a middleware or something like that?

File: projects/desktop/src/components/molecules/languageChooser/LanguageChooserDialog.tsx [hunk 9]
```diff
@@ -21,20 +21,20 @@ export const LanguageChooserDialog = ({open, selectedValue, setLanguage, onClose
...
             </LanguageDialogTitle>
             <LanguageList>
-                {languages.map(language => (
+                {languages.map((language: string): JSX.Element => (
                     <LanguageListItem
                         key={language}
                         data-testid={language}
                         button
-                        onClick={() => setLanguage(language)}
+                        onClick={(): void => setLanguage(language)}
```
Comment (Sylv11) [inline projects/desktop/src/components/molecules/languageChooser/LanguageChooserDialog.tsx:37]: Can you try to only put `setLanguage`?

**Pull request merged**